lib/sarif/base_sarif.rb

```diff
@@ -0,0 +1,42 @@
+module Sarif
+  class BaseSarif
+    DEFAULT_URI = "https://github.com/coinbase/salus".freeze
+  
+    attr_reader :name, :tool, :results, :scan_report
+    def initialize(scan_report)
+      @name = scan_report.scanner_name
+      @scan_report = scan_report
+      @tool = get_tool
+      @results = get_results
+    end
+    
```

Comment:
SARIF Reference: http://docs.oasis-open.org/sarif/sarif/v2.0/csprd01/sarif-v2.0-csprd01.html

Great start.  We will want to build out the generic SARIF a bit further here.


Let's build out the 3.13.1 invocation object.

- 3.13.9 exitCodeDescription 
- 3.13.10 exitSignalName property
- 3.13.11 exitSignalNumber property
- 3.13.13 toolExecutionSuccessful property
Sample from the SARIF spec:
```
{
  "exitCode": 1,
  "exitCodeDescription": "Scan successful; warnings detected.",
  "toolExecutionSuccessful": true
}
```
I'd read through that section of the spec and see if which of these and other's would be a good fit.  Happy to pair on that too if you'd like.  The individual scanning results can follow but we will want to build out to results a little more for our generic SARIF here 👍 

**3.13.20 toolNotification**

```          "toolNotifications": [

            {
              "id": "SAL0001",
              "level": "note",
              "message": {
                "text": "Salus does yet support SARIF details for this scanner"
              }

            }
```
This will give us flexibility to display genuine scan CVE's / issues separately from meta about scanner support.

3.13.21 configurationNotifications could be an option too but I think 3.13.20 toolNotification better reflects the spirit of the SARIF spec here.

Reply:
We probably will a list of toolNotifications that Salus will support.  You can start that list with a single entry if you want as shown above:
```
SALUS_TOOL_NOTIFIICATIONS = { sal0001: "Salus does yet support SARIF details for this scanner" } 
```
or similar

Reply:
Note too as you stand this up, if you find the implementation growing to full construct parts of the SARIF json you can favor composition over inheritance and spin up small dedicated classes for each section. One for toolNotification for example.   Not a have to have, just a note to help you keep the code testable / manageable / SOLID.

Reply:
The toolNotification object is currently available in sarif version 2.0. 

In [sarif version 2.1 ](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317588) there's a toolExecutionNotifications property (3.20.21) which has a similar function.